bundle/src/main/java/com/adobe/acs/commons/wcm/impl/AemEnvironmentIndicatorFilter.java

```diff
@@ -385,4 +425,43 @@ String getTitlePrefix() {
     String[] getExcludedWCMModes() {
         return excludedWCMModes;
     }
+    
+    
+    private class InnerEnvironmentIndicatorFilter implements Filter {
+
+    	String[] excludedWcmModes;
+
+    	public InnerEnvironmentIndicatorFilter(String[] excludedWcmModes) {
+			this.excludedWcmModes = excludedWcmModes;
+		}
+
+		@Override
+		public void init(FilterConfig filterConfig) throws ServletException {
+			// ignore
+			
+		}
+
+		@Override
+		public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+				throws IOException, ServletException {
+
+		    SlingHttpServletRequest req = (SlingHttpServletRequest) request;
+			
+			WCMMode mode = WCMMode.fromRequest(request);
+			if (isDisallowedWcmMode(mode, excludedWcmModes)) {
```

Comment:
`isDisallowedWcmMode` could be moved into this inner filter class, but probably the unit tests cannot compile anymore then, since it's using that strange spy-construction.